catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/XmlModelBuilder.java

```diff
@@ -109,87 +75,194 @@ public JAXBElement getResult() {
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isNotEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::notEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isGreaterThan(boolean matchCase) {
+    beginTerminalType(Mapper::greaterThan);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isGreaterThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::greaterThanOrEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isLessThan(boolean matchCase) {
+    beginTerminalType(Mapper::lessThan);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isLessThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::lessThanOrEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder like(
+      boolean matchCase, String wildcard, String singleChar, String escape) {
+    if (matchCase) {
+      beginTerminalType(Mapper::likeMatchCase);
+      return this;
     }
-    supplierInProgress = new TerminalNodeSupplier(comparisonMapping);
+    beginTerminalType(Mapper::like);
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginPropertyIsLikeType(String operator, boolean matchCase) {
+  public XmlModelBuilder before() {
+    beginTerminalType(Mapper::before);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder after() {
+    beginTerminalType(Mapper::after);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder intersects() {
+    beginTerminalType(Mapper::intersects);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder dwithin(double distance, String units) {
+    beginTerminalType(Mapper::dwithin);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder function(String name) {
     verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    MultiNodeReducer comparisonMapping = TERMINAL_OPS.get(operator);
-    if (comparisonMapping == null) {
-      throw new IllegalArgumentException("Cannot find mapping for like operator: " + operator);
-    }
-    supplierInProgress = new TerminalNodeSupplier(comparisonMapping);
+    verifyTerminalNodeInProgress(); // TODO - Verify this; per schema functions can be predicates
+    supplierInProgress =
+        new UnboundedNodeSupplier<>(
+            args ->
+                FACTORY.createFunction(
+                    new FunctionType().withName(name).withExpression(new ArrayList<>(args))),
+            supplierInProgress);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder property(String name) {
+    setProperty(name);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder value(String value) {
+    setValue(value);
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginBinaryTemporalType(String operator) {
+  public XmlModelBuilder value(boolean value) {
+    // TODO - How can we properly marshal Boolean values without resorting to this?
+    setValue(Boolean.toString(value));
```

Comment:
❗️ Address TODO. This might have been a side effect of JSON parsing. Revisit correct re-hydration of serializable objects into primitive types.

Reply:
JAXB and the XML bindings we use treat everything as a string. Ideally we would leverage the "type" attribute to simply store the type information on a `<Literal/>` element but these bindings were not generated with that in mind, as far as I can tell. For now, some primitive object rehydration code has been added that makes several assumptions but it should suit the needs of the UI. 

The following predicate (& similar) might choke up the system or produce unexpected results: 
```
{
  "type": "=",
  "property": "title",
  "value": "True"
}
```